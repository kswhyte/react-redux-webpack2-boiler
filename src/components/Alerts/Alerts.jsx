import React from 'react';
import PropTypes from 'prop-types';

const Alerts = (props) => {

  Alerts.propTypes = {
    alertType: PropTypes.string,
    message: PropTypes.string,
    closeAlert: PropTypes.string
  };

    return (
        <div className="alerts">

            <div className={`alert alert-${props.alertType}`} role="alert">
            {props.message}<a onClick={props.closeAlert} className="close" aria-label="close">&times;</a>
            </div>
            
        </div>
    );
};

export default Alerts;
