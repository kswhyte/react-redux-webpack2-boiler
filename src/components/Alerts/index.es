import Component from './Alerts';
export default Component;
