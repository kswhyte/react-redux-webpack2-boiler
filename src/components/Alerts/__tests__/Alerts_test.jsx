import React from 'react';
import { storiesOf, action, linkTo } from '@kadira/storybook';
import Alerts from '../Alerts';

import { specs, describe, it } from 'storybook-addon-specifications';

import {mount, shallow} from 'enzyme';
import expect from 'expect';

const successAlert = {
    alertType: 'success',
    message: 'This worked!'
};

const infoAlert = {
    alertType: 'info',
    message: 'Informational Alert!'
};

const warningAlert = {
    alertType: 'warning',
    message: 'Be warned!'
};

const dangerAlert = {
    alertType: 'danger',
    message: 'Something went wrong!'
};


const stories = storiesOf('Alerts', module);

stories.add('Success Alert Options Story', () => {
    const alertsStory = (<Alerts
        {...successAlert}
      />);
    specs(() => describe('Show a successful alert', () => {
        it('Should show a success alert and be green', () => {
            let output = shallow(alertsStory);
            expect(output.find('.alert-success').length).toEqual(1);
        });
    }));
    return alertsStory;
});


stories.add('Info Alert Options Story', () => {
    const alertsStory = (<Alerts
        {...infoAlert}
      />);
    specs(() => describe('Show a info alert', () => {
        it('Should show an info alert and be blue', () => {
            let output = shallow(alertsStory);
            expect(output.find('.alert-info').length).toEqual(1);
        });
    }));
    return alertsStory;
});


stories.add('Warning Alert Options Story', () => {
    const alertsStory = (<Alerts
        {...warningAlert}
      />);
    specs(() => describe('Show a warning alert', () => {
        it('Should show a warning message and be orange', () => {
            let output = shallow(alertsStory);
            expect(output.find('.alert-warning').length).toEqual(1);
        });
    }));
    return alertsStory;
});


stories.add('Danger Alert Options Story', () => {
    const alertsStory = (<Alerts
        {...dangerAlert}
      />);
    specs(() => describe('Show a danger alert', () => {
        it('Should show a danger alert and be red', () => {
            let output = shallow(alertsStory);
            expect(output.find('.alert-danger').length).toEqual(1);
        });
    }));
    return alertsStory;
});