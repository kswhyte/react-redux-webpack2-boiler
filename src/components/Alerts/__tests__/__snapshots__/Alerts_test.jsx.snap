// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Danger Alert Options Story 1`] = `
<div
  className="alerts"
>
  <div
    className="alert alert-danger"
    role="alert"
  >
    Something went wrong!
    <a
      aria-label="close"
      className="close"
      onClick={undefined}
    >
      ×
    </a>
  </div>
</div>
`;

exports[`Info Alert Options Story 1`] = `
<div
  className="alerts"
>
  <div
    className="alert alert-info"
    role="alert"
  >
    Informational Alert!
    <a
      aria-label="close"
      className="close"
      onClick={undefined}
    >
      ×
    </a>
  </div>
</div>
`;

exports[`Success Alert Options Story 1`] = `
<div
  className="alerts"
>
  <div
    className="alert alert-success"
    role="alert"
  >
    This worked!
    <a
      aria-label="close"
      className="close"
      onClick={undefined}
    >
      ×
    </a>
  </div>
</div>
`;

exports[`Warning Alert Options Story 1`] = `
<div
  className="alerts"
>
  <div
    className="alert alert-warning"
    role="alert"
  >
    Be warned!
    <a
      aria-label="close"
      className="close"
      onClick={undefined}
    >
      ×
    </a>
  </div>
</div>
`;
